src/main/java/com/redislabs/redisai/Script.java

```diff
@@ -0,0 +1,140 @@
+    public Script(Device device, String source) {
+        this.device = device;
+        this.source = source;
+        this.tag = "";
+    }
+
+    /**
+     * the device that will execute the model. can be of CPU or GPU
+     *
+     * @param device
+     */
+    public Script(Device device) {
+        this.device = device;
+        this.source = "";
+        this.tag = "";
+    }
+
+    public static Script createScriptFromRespReply(List<?> reply) {
+        Script script = null;
+        String source = null;
+        Device device = null;
+        String tag = null;
+        for (int i = 0; i < reply.size(); i += 2) {
+            String arrayKey = SafeEncoder.encode((byte[]) reply.get(i));
+            switch (arrayKey) {
+                case "source":
+                    source = SafeEncoder.encode((byte[]) reply.get(i + 1));
+                    break;
+                case "device":
+                    String deviceString = SafeEncoder.encode((byte[]) reply.get(i + 1));
+                    device = Device.valueOf(deviceString);
+                    if (device == null) {
+                        throw new JRedisAIRunTimeException("Unrecognized device: " + deviceString);
+                    }
+                    break;
+                case "tag":
+                    tag = SafeEncoder.encode((byte[]) reply.get(i + 1));
+                    break;
+                default:
+                    break;
+            }
+        }
+        if (device != null && source != null) {
+            script = new Script(device, source);
+            if (tag != null) {
+                script.setTag(tag);
+            }
+        }
+        return script;
+    }
+
+    public Device getDevice() {
+        return device;
+    }
+
+    public void setDevice(Device device) {
+        this.device = device;
+    }
+
+    public String getSource() {
+        return source;
+    }
+
+    public void setSource(String source) {
+        this.source = source;
+    }
+
+    public String getTag() {
+        return tag;
+    }
+
+    public void setTag(String tag) {
+        this.tag = tag;
+    }
+
+    /**
+     * Encodes the current script into an AI.SCRIPTSET command to be store in RedisAI Server
+     *
+     * @param key name of key to store the Script
+     * @return
+     */
+    protected List<byte[]> getScriptSetCommandBytes(String key) {
+        List<byte[]> args = new ArrayList<>();
+        args.add(SafeEncoder.encode(key));
+        args.add(device.getRaw());
+        args.add(Keyword.SOURCE.getRaw());
+        args.add(SafeEncoder.encode(source));
+        return args;
+    }
+
+    /**
+     * sets the Script source give a filePath
+     *
+     * @param filePath
+     * @throws IOException
+     */
+    public void readSourceFromFile(String filePath) throws IOException {
+        this.source = Files.readAllLines(Paths.get(filePath), StandardCharsets.UTF_8)
+                .stream()
+                .collect(Collectors.joining("\n")) + "\n";
```

Comment:
@filipecosta90 why do you need the extra "\n"?

Reply:
This code I only moved from the place but having \n\n won't be impact scriptset and not having \n will not work - I believe was added to prevent scripts not containing the newline at the end to fail. 
WDYT, should we keep this? the other clients don't add it, so imho we should remove the extra \n.

Reply:
just confirmed that `Files.readAllLines` is not preserving the last `\n` since it reads a line up until the line terminator and throws it away ( since the last line only contains it we need to add it again )